source/common/ur_pool_manager.hpp

```diff
@@ -158,63 +194,83 @@ inline bool pool_descriptor::operator==(const pool_descriptor &other) const {
         }
     }
 
+    if (lhs.allocLocation.has_value()) {
```

Comment:
nit: I think you can simplify it to: `if (lhs.allocLocation != rhs.allocLocation) return false;`

Reply:
Ah, nevermind, `pNext` might be different. I'm wondering if we should perhaps implement operator== for ur_usm_alloc_location_desc_t ...